tests/components/islamic_prayer_times/test_sensor.py

```diff
@@ -67,120 +36,22 @@ async def test_islamic_prayer_times_multiple_sensors(hass):
     ]
 
     with patch(
-        "homeassistant.components.islamic_prayer_times.sensor.PrayerTimesCalculator"
-    ) as PrayerTimesCalculator:
-        PrayerTimesCalculator.return_value.fetch_prayer_times.return_value = (
-            PRAYER_TIMES
+        "prayer_times_calculator.PrayerTimesCalculator.fetch_prayer_times",
+        return_value=PRAYER_TIMES,
+    ):
+        config_entry = MockConfigEntry(
+            domain=islamic_prayer_times.DOMAIN, data={}, options=MOCK_OPTIONS
         )
+        config_entry.add_to_hass(hass)
+        await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
 
-        config = {
-            "sensor": {"platform": "islamic_prayer_times", "sensors": multiple_sensors}
-        }
-
-        assert await async_setup_component(hass, "sensor", config) is True
-
-        for sensor in multiple_sensors:
-            entity_id = ENTITY_ID_FORMAT.format(sensor)
-            entity_id_name = sensor.capitalize()
-            pt_dt = get_prayer_time_as_dt(PRAYER_TIMES[entity_id_name])
-            state = hass.states.get(entity_id)
-            assert state.state == pt_dt.isoformat()
-            assert state.name == entity_id_name
-
-
-async def test_islamic_prayer_times_with_calculation_method(hass):
-    """Test Islamic prayer times configuration with calculation method."""
-    sensors = ["fajr", "maghrib"]
-
-    with patch(
-        "homeassistant.components.islamic_prayer_times.sensor.PrayerTimesCalculator"
-    ) as PrayerTimesCalculator:
-        PrayerTimesCalculator.return_value.fetch_prayer_times.return_value = (
-            PRAYER_TIMES
-        )
-
-        config = {
-            "sensor": {
-                "platform": "islamic_prayer_times",
-                "calculation_method": "mwl",
-                "sensors": sensors,
-            }
-        }
-
-        assert await async_setup_component(hass, "sensor", config) is True
-
-        for sensor in sensors:
+        for sensor in min_config_sensors:
             entity_id = ENTITY_ID_FORMAT.format(sensor)
             entity_id_name = sensor.capitalize()
-            pt_dt = get_prayer_time_as_dt(PRAYER_TIMES[entity_id_name])
+            pt_dt = hass.data[islamic_prayer_times.DOMAIN].get_prayer_time_as_dt(
```

Comment:
It's not so robust to depend on the integration details here. It would be better to define what the data is that will be returned by the client, ie `pt_dt`, before the test and then just assert that the state corresponds to that.